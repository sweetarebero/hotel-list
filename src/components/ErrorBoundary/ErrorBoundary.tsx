import React, { Component, ReactNode } from "react";

interface Props {
  children: ReactNode;
}

interface State {
  hasError: boolean;
  message: string;
}

class ErrorBoundary extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      hasError: false,
      message: "",
    };
  }

  public static getDerivedStateFromError(error: Error): State {
    /* Update state so the next render will show the fallback UI */
    return { hasError: true, message: error.message };
  }

  public render() {
    const { hasError } = this.state;
    const { children } = this.props;

    return hasError ? (
      <h1 data-testid="errorboundary">
        Oops! something went wrong, please try later.
      </h1>
    ) : (
      children
    );
  }
}

export default ErrorBoundary;
