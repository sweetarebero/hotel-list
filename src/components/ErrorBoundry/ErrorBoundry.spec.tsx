import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import ErrorBoundary from "./ErrorBoundry";

describe("Error Boundary render", () => {
  it("renders correctlty", () => {
    const { container } = render(
      <ErrorBoundary>
        <div>children</div>
      </ErrorBoundary>
    );

    expect(container).toMatchSnapshot();
  });
});

describe("Error Boundary error verification", () => {
  test("Error Boundary", () => {
    const ThrowError = () => {
      throw new Error("Test");
    };
    render(
      <ErrorBoundary>
        <ThrowError />
      </ErrorBoundary>
    );
    expect(screen.getByTestId("errorboundary")).toBeVisible();
  });
});
