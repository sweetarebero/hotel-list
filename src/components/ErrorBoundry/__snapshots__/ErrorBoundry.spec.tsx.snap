// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary render renders correctlty 1`] = `
<div>
  <div>
    children
  </div>
</div>
`;
